flood_adapt/adapter/sfincs_adapter.py

```diff
@@ -1,210 +1,650 @@
         )
+        if isinstance(forcing, IRainfall):
+            self._add_forcing_rain(forcing)
+        elif isinstance(forcing, IWind):
+            self._add_forcing_wind(forcing)
+        elif isinstance(forcing, IDischarge):
+            self._add_forcing_discharge(forcing)
+        elif isinstance(forcing, IWaterlevel):
+            self._add_forcing_waterlevels(forcing)
+        else:
+            self.logger.warning(
+                f"Skipping unsupported forcing type {forcing.__class__.__name__}"
+            )
 
-    def add_wind_forcing_from_grid(self, ds: xr.DataArray):
-        """Add spatially varying wind forcing to sfincs model.
+    def add_measure(self, measure: IMeasure):
+        """Get measure data and add it to the sfincs model."""
+        self.logger.info(
+            f"Adding {measure.__class__.__name__.capitalize()} to the SFINCS model..."
+        )
 
-        Parameters
-        ----------
-        ds : xr.DataArray
-            Dataarray which should contain:
-            - wind_u: eastward wind velocity [m/s]
-            - wind_v: northward wind velocity [m/s]
-            - spatial_ref: CRS
-        """
-        if ds["lon"].min() > 180:
-            ds["lon"] = ds["lon"] - 360
-        self.sf_model.setup_wind_forcing_from_grid(wind=ds)
+        if isinstance(measure, FloodWall):
+            self._add_measure_floodwall(measure)
+        elif isinstance(measure, GreenInfrastructure):
+            self._add_measure_greeninfra(measure)
+        elif isinstance(measure, Pump):
+            self._add_measure_pump(measure)
+        else:
+            self.logger.warning(
+                f"Skipping unsupported measure type {measure.__class__.__name__}"
+            )
 
-    def add_pressure_forcing_from_grid(self, ds: xr.DataArray):
-        """Add spatially varying barometric pressure to sfincs model.
+    def add_projection(self, projection: IProjection):
+        """Get forcing data currently in the sfincs model and add the projection it."""
+        self.logger.info("Adding Projection to the SFINCS model...")
+        phys_projection = projection.get_physical_projection()
 
-        Parameters
-        ----------
-        ds : xr.DataArray
-            Dataarray which should contain:
-            - press: barometric pressure [Pa]
-            - spatial_ref: CRS
-        """
-        self.sf_model.setup_pressure_forcing_from_grid(press=ds)
+        if phys_projection.attrs.sea_level_rise:
+            self.logger.info(
+                f"Adding sea level rise ({phys_projection.attrs.sea_level_rise}) to SFINCS model."
+            )
+            self.waterlevels += phys_projection.attrs.sea_level_rise.convert(
+                us.UnitTypesLength.meters
+            )
 
-    def add_precip_forcing_from_grid(self, ds: xr.DataArray):
-        """Add spatially varying precipitation to sfincs model.
+        # ? phys_projection.attrs.subsidence
```

Comment:
I asked about this in yesterday's meeting since I couldnt find any other reference to subsidence in the codebase, except for this comment.

There used to be this subsidence parameter in the projection, however noone used it and it was removed from the gui / backend since it was implemented the same as slr (slr means water go up, subsidence means house go down)

I'll remove the comment